pulpcore/cli/file/repository.py

```diff
@@ -64,14 +64,14 @@ def repository(ctx: click.Context, pulp_ctx: PulpContext, repo_type: str) -> Non
 
 lookup_options = [href_option, name_option]
 nested_lookup_options = [repository_href_option, repository_option]
-create_options = [
-    click.option("--name", required=True),
-    click.option("--description"),
-    click.option("--remote", callback=_remote_callback),
-]
 update_options = [
     click.option("--description"),
     click.option("--remote", callback=_remote_callback),
+    click.option("--manifest"),
+    click.option("--autopublish/--no-autopublish", default=None),
```

Comment:
IMO the default should simply be --no-autopublish.

Reply:
This is an update_option. So if not specified, we should def _not_ set (and then update) this option to `False`.
I think having `None` here, keeps it from being touched on update calls, and on create calls i'd rather let the server decide on the default.

Reply:
Ah, I believe I misunderstood what it does then

Reply:
I agree it should default to `--no-autopublish`. The dilemma is that this will cause `autopublish` to always be set in the requests which won't work for older versions of pulp_file/pulp_rpm and we support 5 versions back.

I could either:

1. Check the version of pulp_file and if it's less than 1.7.0, silently strip out the autopublish key/value from the request.
2. Keep the default as None which won't set autopublish in requests unless the user explicitly sets `--autopublish` or `--no-autopublish`. I could add a comment for us to change the default this once we drop support for 1.7.0.

Thoughts?

Reply:
Welp, ignore my comment. I am too slow.